### PR TITLE
Add support for PVS-Studio

### PR DESCRIPTION
**Pull request by ValeryKomarov**

### CODE REVIEW

**Comment by SvyatoslavRazmyslov** [comment]
@uhafner ,
Hi, can you check and merge the PR, please?

File: src/main/java/io/jenkins/plugins/analysis/warnings/PVSStudio.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        // empty constructor required for stapler
+    }
+
+    @Override
+    public IssueParser createParser() {
+        return new PVSStudioParser();
+    }
+
+    /** Provides the labels for the static analysis tool. */
+    private static class LabelProvider extends IconLabelProvider {
```
Comment (uhafner) [inline src/main/java/io/jenkins/plugins/analysis/warnings/PVSStudio.java]: Can't you directly use `IconLabelProvider`?

File: src/main/java/io/jenkins/plugins/analysis/warnings/PVSStudio.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        /** Creates the descriptor instance. */
+        public Descriptor() {
+            super(ID);
+        }
+
+        @NonNull
+        @Override
+        public String getDisplayName() {
+            return "PVS-Studio";
+        } // title page
```
Comment (uhafner) [inline src/main/java/io/jenkins/plugins/analysis/warnings/PVSStudio.java]: Remove comment

File: src/main/java/io/jenkins/plugins/analysis/warnings/PVSStudio.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+        @Override
+        public StaticAnalysisLabelProvider getLabelProvider() {
+            return new LabelProvider(ID, "PVS-Studio", "pvs");
```
Comment (uhafner) [inline src/main/java/io/jenkins/plugins/analysis/warnings/PVSStudio.java]: See above, use `new IconLabelProvider`

File: src/main/java/io/jenkins/plugins/analysis/warnings/PVSStudio.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        }
+
+        @Override
+        public String getPattern() {
+            return "**/**.plog";
+        }
+    }
```
Comment (uhafner) [inline src/main/java/io/jenkins/plugins/analysis/warnings/PVSStudio.java]: Do you also have a URL? Then you can overwrite `getUrl`?

File: src/main/java/io/jenkins/plugins/analysis/warnings/PVSStudio.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+/**
+ * Provides a parser and customized messages for PVS-Studio static analyzer.
+ *
+ * @author PVS-Studio Team
+ */
+public class PVSStudio extends ReportScanningTool {
+
+    private static final long serialVersionUID = -1114828406964963020L;
+    private static final String ID = "pvs-studio"; // history chart title
```
Comment (uhafner) [inline src/main/java/io/jenkins/plugins/analysis/warnings/PVSStudio.java]: Can you please add a license note for the icon to [LICENSE.txt](https://github.com/jenkinsci/warnings-ng-plugin/blob/master/src/main/webapp/icons/LICENSE.txt)?

File: src/main/java/io/jenkins/plugins/analysis/warnings/PVSStudio.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+import io.jenkins.plugins.analysis.core.model.StaticAnalysisLabelProvider;
+
+import edu.hm.hafner.analysis.parser.pvsstudio.PVSStudioParser;
+
+/**
+ * Provides a parser and customized messages for PVS-Studio static analyzer.
+ *
+ * @author PVS-Studio Team
+ */
+public class PVSStudio extends ReportScanningTool {
```
Comment (uhafner) [inline src/main/java/io/jenkins/plugins/analysis/warnings/PVSStudio.java]: Can you please also add a small test in the style of [ParsersITest.java](https://github.com/jenkinsci/warnings-ng-plugin/blob/master/src/test/java/io/jenkins/plugins/analysis/warnings/ParsersITest.java#L114) that counts the warnings of a single file? You can reuse the file of the analysis-model test.

**Comment by uhafner** [comment]
Please also run ToolsLister test case so that the list of supported formats will be updated...

**Comment by oleg-nenashev** [comment]
Thanks a lot for the feature @ValeryKomarov !

**Pull request merged**